src/osdp_cp.c

```diff
@@ -1130,9 +1130,11 @@ osdp_t *osdp_cp_setup(int num_pd, osdp_pd_info_t *info, uint8_t *master_key)
 		}
 	}
 
-	if (master_key == NULL && scbk_count != num_pd) {
-		LOG_WRN("Master key / SCBK not passed; SC Disabled.");
-		SET_FLAG(ctx, FLAG_SC_DISABLED);
+	if (master_key == NULL) {
+		if (scbk_count != num_pd) {
+			LOG_WRN("Master key / SCBK not passed; SC Disabled.");
+			SET_FLAG(ctx, FLAG_SC_DISABLED);
+		}
```

Comment:
Looks like this also fixes a bug. If `master_key == NULL` and `scbk_count == num_pd`, we copy the master_key anyway. So can you reword your commit message to have something like "Fix: you reason" so it will make it into the right section in the next release notes?

Reply:
AND.. In the case where we do take the master key we need to do 2 things:
 - Check for `OSDP_FLAG_ENFORCE_SECURE` and not do it.
 - Loop over all PDs and clear the `PD_FLAG_HAS_SCBK` flag as src/osdp_sc.c line 42 does a check on this flag before computing the SCBK with master key.

Reply:
Just updated the commit message.  You can, of course, feel to change it you can see a way to improve it.

Reply:
I was thinking the case of taking the master key was okay regarding the flags and security and an additional code change wasn't needed relating to these, at least at this time.  I'll think some more though and maybe create an issue for discussion on a related topic.